_opensearch/rest-api/analyze-apis/perform-text-analysis.md

```diff
@@ -0,0 +1,670 @@
+
+The following query parameter is required.
+
+| Field | Data Type | Description
+:--- | :--- | :---
+text | String or Array of Strings | Text to analyze. If you provide an array of strings, it is analyzed as a multi-value field.
+
+#### Sample requests
+
+[Analyze array of text strings](#analyze-array-of-text-strings)
+
+[Apply a built-in analyzer](#apply-a-built-in-analyzer)
+
+[Apply a custom analyzer](#apply-a-custom-analyzer)
+
+[Apply a custom transient analyzer](#apply-a-custom-transient-analyzer)
+
+[Specify an index](#specify-an-index)
+
+[Derive the analyzer from an index field](#derive-the-analyzer-from-an-index-field)
+
+[Specify a normalizer](#specify-a-normalizer)
+
+[Get token details](#get-token-details)
+
+[Set a token limit](#set-a-token-limit)
+
+#### Analyze array of text strings
+
+When you pass an array of strings to the `text` field it is analyzed as a multi-value field.
+
+````json
+GET /_analyze
+{
+  "analyzer" : "standard",
+  "text" : ["first array element", "second array element"]
+}
+````
+
+The previous request returns the following fields:
+
+````json
+{
+  "tokens" : [
+    {
+      "token" : "first",
+      "start_offset" : 0,
+      "end_offset" : 5,
+      "type" : "<ALPHANUM>",
+      "position" : 0
+    },
+    {
+      "token" : "array",
+      "start_offset" : 6,
+      "end_offset" : 11,
+      "type" : "<ALPHANUM>",
+      "position" : 1
+    },
+    {
+      "token" : "element",
+      "start_offset" : 12,
+      "end_offset" : 19,
+      "type" : "<ALPHANUM>",
+      "position" : 2
+    },
+    {
+      "token" : "second",
+      "start_offset" : 20,
+      "end_offset" : 26,
+      "type" : "<ALPHANUM>",
+      "position" : 3
+    },
+    {
+      "token" : "array",
+      "start_offset" : 27,
+      "end_offset" : 32,
+      "type" : "<ALPHANUM>",
+      "position" : 4
+    },
+    {
+      "token" : "element",
+      "start_offset" : 33,
+      "end_offset" : 40,
+      "type" : "<ALPHANUM>",
+      "position" : 5
+    }
+  ]
+}
+````
+
+#### Apply a built-in analyzer
+
+If you omit the `index` path parameter, you can apply any of the built-in analyzers to the text string.
+
+The following request analyzes text using the `standard` built-in analyzer:
+
+````json
+GET /_analyze
+{
+  "analyzer" : "standard",
+  "text" : "OpenSearch text analysis"
+}
+````
+
+The previous request returns the following fields:
+
+````json
+{
+  "tokens" : [
+    {
+      "token" : "opensearch",
+      "start_offset" : 0,
+      "end_offset" : 10,
+      "type" : "<ALPHANUM>",
+      "position" : 0
+    },
+    {
+      "token" : "text",
+      "start_offset" : 11,
+      "end_offset" : 15,
+      "type" : "<ALPHANUM>",
+      "position" : 1
+    },
+    {
+      "token" : "analysis",
+      "start_offset" : 16,
+      "end_offset" : 24,
+      "type" : "<ALPHANUM>",
+      "position" : 2
+    }
+  ]
+}
+````
+
+#### Apply a custom analyzer
+
+You can create your own analyzer and specify it in an analyze request.
```

Comment:
Back in line 25, this was "analyzer requests". Is "analyzer request" accurate here?

Reply:
corrected above